### PR TITLE
fix: move list quorum ENV to config

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: move list quorum ENV to config

## Motivation and Context
allow common config and help

## How to test this PR?
Supported values are now `optimal`, `reduced`, `disk`, and `strict` with `optimal` being the default.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

File: cmd/config/api/api.go [hunk 1]
```diff
@@ -94,6 +102,24 @@ func (sCfg *Config) UnmarshalJSON(data []byte) error {
...
+// GetListQuorum interprets list quorum values and returns appropriate
+// acceptable quorum expected for list operations
+func (sCfg Config) GetListQuorum() int {
+	switch sCfg.ListQuorum {
+	case "optimal":
+		return 3
+	case "reduced":
+		return 2
+	case "disk":
+		// least possible value, generally meant for testing.
```
Comment (klauspost) [inline cmd/config/api/api.go]:
```suggestion
		// smallest possible value, generally meant for testing.
```

File: cmd/config/api/api.go [hunk 2]
```diff
@@ -94,6 +102,24 @@ func (sCfg *Config) UnmarshalJSON(data []byte) error {
...
+		return 1
+	case "strict":
+		return -1
+	}
+	// Defaults to 50% of totalDrives per set.
+	return -1
```
Comment (klauspost) [inline cmd/config/api/api.go]: Let's make the default consistent.
```suggestion
	// Defaults to 3 drives per set.
	return 3
```

File: cmd/globals.go [hunk 3]
```diff
@@ -159,7 +159,7 @@ var (
 	globalBucketTargetSys    *BucketTargetSys
 	// globalAPIConfig controls S3 API requests throttling,
 	// healthcheck readiness deadlines and cors settings.
-	globalAPIConfig apiConfig
+	globalAPIConfig = apiConfig{listQuorum: -1}
```
Comment (klauspost) [inline cmd/globals.go]:
```suggestion
	globalAPIConfig = apiConfig{listQuorum: 3}
```

File: cmd/config/storageclass/storage-class.go [hunk 4]
```diff
@@ -38,7 +38,7 @@ const (
 
 	// Valid values are "write" and "read-write"
 	DMAWrite     = "write"
-	DMAReadWrite = "read-write"
+	DMAReadWrite = "read+write"
```
Comment (klauspost) [inline cmd/config/storageclass/storage-class.go:41]: This or comment above is wrong.

**Review by klauspost — COMMENTED** [review]
Just some minor stuff I noticed.

File: cmd/config/storageclass/storage-class.go (hunk 4 above)
Comment (harshavardhana) [reply cmd/config/storageclass/storage-class.go:41]: I changed it let me fix the comment as well.

File: cmd/config/storageclass/storage-class.go [hunk 5]
```diff
@@ -38,7 +38,7 @@ const (
 
 	// Valid values are "write" and "read-write"
```
Comment (harshavardhana) [inline cmd/config/storageclass/storage-class.go]:
```suggestion
	// Valid values are "write" and "read+write"
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#385488c3-5545-4e98-92a0-45a0f3731146) |
<a name="385488c3-5545-4e98-92a0-45a0f3731146"></a>
### 10804-42e5e67/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31763
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4a2fe235ec2d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 23 seconds
(4/15) Running aws-sdk-ruby tests ... done in 21 seconds
(5/15) Running awscli tests ... done in 2 minutes and 46 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 4 minutes and 18 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 35 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 48 seconds
{
  "args": {
    "bucketName": "minio-go-test-ir9wdrroljjyys2n",
    "objectName": "test-object",
    "opts": "",
    "size": -1
  },
  "duration": 1182,
  "function": "PutObject(bucketName, objectName, reader,size,opts)",
  "message": "Unexpected size",
  "name": "minio-go: testPutObjectStreaming",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 10 minutes and 15 seconds
(10/15) Running minio-js tests ... FAILED in 47 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 80,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:682:10) at endReadableNT (_stream_readable.js:1252:12) at processTicksAndRejections (internal/process/task_queues.js:80:21)"
}
(10/15) Running minio-py tests ... FAILED in 18 minutes and 18 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-ff60591a-3ed0-4a83-8b52-0c41bcf85737",
    "object_name": "cd7f47fc-9c75-40d3-bc00-a5623e7fb483"
  },
  "message": "Sha-sum mismatch on multi-threaded put and get objects",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 145, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1666, in test_thread_safe\n    raise exceptions[0]\n  File \"/mint/run/core/minio-py/tests.py\", line 1631, in get_object_and_check\n    'Sha-sum mismatch on multi-threaded put and '\nValueError: Sha-sum mismatch on multi-threaded put and get objects\n",
  "duration": 35840
}
(10/15) Running s3cmd tests ... done in 2 minutes and 22 seconds
(11/15) Running s3select tests ... done in 39 seconds
(12/15) Running security tests ... done in 0 seconds

Executed 12 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Pull request merged**